### PR TITLE
chore: clean code DefaultChannelPipeline add method

### PR DESCRIPTION
**Pull request by Roiocam**
Motivation:

DefaultChannelPipeline has some duplicate code on there, it can be simplified using templates

Modification:

The new `internalAdd` method is used by `addFirst`, `addLast`, `addBefore` and `addAfter`

Result:

Clean code

### CODE REVIEW

File: transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java [hunk 1]
```diff
@@ -151,16 +152,21 @@ public final ChannelPipeline addFirst(String name, ChannelHandler handler) {
         return addFirst(null, name, handler);
     }
 
-    @Override
-    public final ChannelPipeline addFirst(EventExecutorGroup group, String name, ChannelHandler handler) {
+    private ChannelPipeline internalAdd(EventExecutorGroup group, String name, ChannelHandler handler, String baseName, BiConsumer<AbstractChannelHandlerContext,AbstractChannelHandlerContext> addMethod){
```
Comment (normanmaurer) [inline transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java:155]: 4.1 can not use `BiConsumer` as its java8.

Comment (Roiocam) [reply transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java]: thanks for your review, I have replaced it with an enum(the strategy pattern may be too heavy for it). 

what do you think about this PR change idea, it is probably not required, just cleaner for me.

**Comment by normanmaurer** [comment]
@Roiocam did you sign our ical ? https://netty.io/s/icla

**Comment by Roiocam** [comment]
> @Roiocam did you sign our ical ? [netty.io/s/icla](https://netty.io/s/icla)

Yes, I found no format plugin, so I manually added a line break to fix the CI failed.

**Review by normanmaurer — APPROVED** [review]

**Comment by normanmaurer** [comment]
@Roiocam thanks!

**Pull request merged**